sirepo/package_data/static/js/raydata.js

```diff
@@ -30,70 +33,82 @@ SIREPO.app.factory('raydataService', function(appState, requestSender, simulatio
 	return cols.length > 0 ? [firstColHeading].concat(cols) : [];
     };
 
-    self.getScansInfo = function(scanUuids, successCallback, options) {
-	function maybeGetOnlyOneScan(scans) {
-	    return s ? scans[0] : scans;
-	}
-
-	let s = false;
-	if (typeof scanUuids === 'string') {
-	    scanUuids = [scanUuids];
-	    s = true;
-	} else if (! Array.isArray(scanUuids)) {
-	    scanUuids = Object.keys(scanUuids);
-	}
-	if (scanUuids.every((e) => {
+    self.getScansInfo = function(successCallback, options) {
+	const s = Object.keys(appState.models.scans.selected);
+	if (s.every((e) => {
 	    // POSIT: If start is present so are the other fields we need
 	    return e in (simulationDataCache.scans || {}) && simulationDataCache.scans[e].start;
 	})) {
 	    successCallback(
-		maybeGetOnlyOneScan(
-		    scanUuids.map((u) => simulationDataCache.scans[u]
-		).sort((a, b) => a.start > b.start ? 1 : -1)),
+		s.map(
+		    (u) => simulationDataCache.scans[u]
+		).sort((a, b) => a.start > b.start ? 1 : -1),
```

Comment:
more conventional is to return the difference of the values

Reply:
Thanks didn't know about that. Also, caught a bug where sort didn't happen after the request to the server. e46a18f00